src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -3330,7 +3343,15 @@ private void EmitCallingForBaseConstructor(int thisClassIndex, ProtoCore.AST.Ass
                 if (bidx != Constants.kInvalidIndex)
                 {
                     baseConstructorName = core.ClassTable.ClassNodes[bidx].Name;
-                    int cidx = core.ClassTable.ClassNodes[bidx].ProcTable.IndexOf(baseConstructorName, argTypeList);
+                    int cidx = core.ClassTable.ClassNodes[bidx].ProcTable.GetFunctionBySignature(new ProcedureMatchOptions()
+                    {
+                        FunctionName = baseConstructorName,
+                        ParameterTypes = argTypeList,
+                        ExcludeAutoGeneratedThisProc = true,
+                        IsConstructor = true,
+                        ExactMatchWithNumArgs = false,
+                        ExactMatchWithArgTypes = false
+                    }, out ProcedureNode dummy);
```

Comment:
why dummy?

Reply:
because it is not used and an output param cannot be optional.
The output param is sometimes useful (that's why I added it)

Reply:
I can use discard "_"